superflore/test_integration/gentoo/build_base.py

```diff
@@ -26,18 +26,35 @@ def __init__(
         self.container = Docker()
         self.container.pull(image_owner, image_name)
         self.package_list = dict()
+        self.default_upstream = True
```

Comment:
Instead of adding a field to the class which might be modified  by hand without changing upstream behavior does it make sense to always invoke set_upstream and change its behavior based on whether or not an upstream is provided or not?

This seems like a case where we could share data by passing messages rather than passing messages by sharing data.